inference-engine/ie_bridges/python/tests/test_InferRequest.py

```diff
@@ -362,6 +363,24 @@ def execute(self, input_data):
     del ie_core
 
 
+def test_async_infer_wait_while_callback_will_not_finish(device):
+    def callback(status, callback_status):
+        time.sleep(0.5)
```

Comment:
Why do we need that sleep here?

Reply:
To make sure that InferRequest.wait waits callback function. Without sleep, result of this test will be dependent from Race condition.

Reply:
That should not be a case here... Callback should bring GIL for itself and put the data inside `callback_status['finished']` and only after that release the GIL back to be grabbed by main python thread. If there is some kind of the race condition in callback (that should fully execute on python side with full access to python world) that probably means lock is not held during execution on this callback and would also behave unexpectedly in other cases. Can you investigate the case further?

Reply:
Yes, as you can see here:
https://github.com/openvinotoolkit/openvino/blob/809bb8af2a4347ae22aba0a99bb248aba9c65b0c/inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx#L1050: 
callback function locks the GIL before starting. But there is no guarantees that main python will lock the GIL first here:
https://github.com/openvinotoolkit/openvino/blob/809bb8af2a4347ae22aba0a99bb248aba9c65b0c/inference-engine/ie_bridges/python/tests/test_InferRequest.py#L381
I tested it, without sleep this test passes on master, so we need sleep, to verify that wait blocks main python thread until callback completes